sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -337,10 +336,11 @@ class SidechainState private[horizen] (stateStorage: SidechainStateStorage,
         throw new Exception("Amounts sum of CoinsBoxes is incorrect. " +
           s"ClosedBox amount - $closedCoinsBoxesAmount, NewBoxesAmount - $newCoinsBoxesAmount, Fee - ${tx.fee()}")
 
+      lazy val isForgerOpen = isForgingOpen()
       newBoxes
         .filter(box => box.isInstanceOf[ForgerBox])
         .foreach(forgerBox => {
-          if (params.restrictForgers && !openForger()) {
+          if (params.restrictForgers && !isForgerOpen) {
```

Comment:
`if (params.restrictForgers && !isForgerOpen) {`
can be replaced just with `if(!isForgerOpen) {`
since now `params.restrictForgers` is checked inside `isForgerOpen` method.

Reply:
True, thanks!